internal/controllers/mbsc_reconciler.go

```diff
@@ -0,0 +1,200 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	kmmv1beta1 "github.com/kubernetes-sigs/kernel-module-management/api/v1beta1"
+	"github.com/kubernetes-sigs/kernel-module-management/internal/api"
+	"github.com/kubernetes-sigs/kernel-module-management/internal/buildsign"
+	"github.com/kubernetes-sigs/kernel-module-management/internal/kernel"
+	"github.com/kubernetes-sigs/kernel-module-management/internal/mbsc"
+	"github.com/kubernetes-sigs/kernel-module-management/internal/utils"
+	v1 "k8s.io/api/core/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const MBSCReconcilerName = "MBSCReconciler"
+
+// mbscReconciler reconciles a Module object
+type mbscReconciler struct {
+	reconHelperAPI mbscReconcilerHelperAPI
+}
+
+func NewMBSCReconciler(
+	client client.Client,
+	buildSignAPI buildsign.Manager,
+	mbscAPI mbsc.MBSC,
+) *mbscReconciler {
+	reconHelperAPI := newMBSCReconcilerHelper(client, buildSignAPI, mbscAPI)
+	return &mbscReconciler{
+		reconHelperAPI: reconHelperAPI,
+	}
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *mbscReconciler) SetupWithManager(mgr ctrl.Manager, kernelLabel string) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&kmmv1beta1.ModuleBuildSignConfig{}).
+		Owns(&v1.Pod{}).
+		Named(MBSCReconcilerName).
+		Complete(
+			reconcile.AsReconciler[*kmmv1beta1.ModuleBuildSignConfig](mgr.GetClient(), r),
+		)
+}
+
+//+kubebuilder:rbac:groups=kmm.sigs.x-k8s.io,resources=modulebuildsignconfigs,verbs=get;list;watch;update;patch;create
+//+kubebuilder:rbac:groups=kmm.sigs.x-k8s.io,resources=modulebuildsignconfigs/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups="core",resources=secrets,verbs=get;list;watch
+//+kubebuilder:rbac:groups="core",resources=configmaps,verbs=get;list;watch
+//+kubebuilder:rbac:groups="core",resources=pods,verbs=create;list;watch;delete
+
+// Reconcile lists all nodes and looks for kernels that match its mappings.
+// For each mapping that matches at least one node in the cluster, it creates a DaemonSet running the container image
+// on the nodes with a compatible kernel.
+func (r *mbscReconciler) Reconcile(ctx context.Context, mbscObj *kmmv1beta1.ModuleBuildSignConfig) (ctrl.Result, error) {
+	res := ctrl.Result{}
+
+	logger := log.FromContext(ctx)
+
+	if mbscObj.GetDeletionTimestamp() != nil {
+		// [TODO] delete build/sign pods
```

Comment:
Why don't we do it as part of this PR?

Reply:
we have the same issue in the MIC controller also. I would prefer to split PRs between 2 different flows:
1) the normal flow
2) deletion flow